File: 485. Max Consecutive Ones/solution.cpp
# include<algorithm>
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0;
        int result = 0;
        // int ans = 0;
        if (nums.empty())
            return 0;
        else {
            count = 0;
            result = 0;
            for (int i = 0; i < nums.size(); i++) { // Traversal
                if (nums [i] == 1)
                    count ++;
                else {
                    result = max(result, count);
                    count = 0;
                }
            }
            // ans = max(result, count);
            return max(result, count);
        }
    }
};

File: 485. Max Consecutive Ones/solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int result = 0;
        int count = 0;
        int ans = 0;
        int size = nums.length;
        if (size == 0)
            return ans;
        else {
            for (int i = 0; i < size; i ++) {
            if (nums[i] == 1) {
                count ++;
            }
            else {
                result = Math.max(result, count);
                count = 0;
            }
        }
        }
        ans = Math.max(result, count);
        return ans;
    }
}
